pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/AggInfo.java

```diff
@@ -41,24 +40,47 @@
  * Example usage:
  * Map<String, String> params = new HashMap<>();
  * params.put("window", "5m");
- * AggInfo aggInfo = new AggInfo("rate", params);
+ * AggInfo aggInfo = new AggInfo("rate", true, params);
  */
 public class AggInfo {
   private final String _aggFunction;
+  /**
+   * Denotes whether an aggregate is partial or full. When returning the logical plan, language developers must not
+   * set this to true. This is used during Physical planning, and Pinot may set this to true if the corresponding
+   * aggregate node is not guaranteed to have the full data. In such cases, the physical plan will always add a
+   * complimentary full aggregate.
+   * <p>
+   *  TODO(timeseries): Ideally we should remove this from the logical plan completely.
+   * </p>
+   */
+  private final boolean _isPartial;
```

Comment:
Based on the _isPartial flag Users are expected to run SeriesBuilderoperations. For example if _isPartial = true merge series based on function which would produce final result for leaf phase. 
Am I correct?

Reply:
Yup. Note that you should set isPartial to false in the logical plan that you return. Pinot will automatically create a partial AggInfo when the execution is split between the brokers and servers.